WcaOnRails/app/models/competition.rb

```diff
@@ -807,4 +821,20 @@ def serializable_hash(options = nil)
       organizers: organizers,
     }
   end
+
+  # See https://github.com/thewca/worldcubeassociation.org/wiki/WCA-Competition-JSON-Format
+  def big_json
+    big_json_db = JSON.parse(super || "{}")
+    big_json_db.merge(
+      "formatVersion" => "WCA Competition 0.3",
+      "competitionId" => id,
+      "scrambleProgram" => nil,
+      "resultsProgram" => nil,
+    )
+  end
+
+  def big_json=(new_big_json)
+    big_json = self.big_json.merge(new_big_json)
+    super(JSON.generate(big_json))
```

Comment:
You're right, there's a lot of room for validations to occur in this `big_json=` method (and I do hope that once the big json spec is closer to finalized, we'll start to implement those validations).

For everyone following along, Philippe and I already discussed this: we don't have to worry about the user changing something they're not supposed to (such as the competition id), because in our `def big_json` method, we read the json from our database, and then clobber it with whatever structured data we already have. For instance, @viroulep is going to add registration data in there as soon as this PR gets merged up =)